sycl/include/syclcompat/syclcompat.hpp

```diff
@@ -22,6 +22,24 @@
 
 #pragma once
 
+// MSVC ignores [[deprecated]] attribute on namespace unless compiled with
+// /W3 or above.
+#ifdef _MSC_VER
+#define __SYCLCOMPAT_STRINGIFY(x) #x
+#define __SYCLCOMPAT_TOSTRING(x) __SYCLCOMPAT_STRINGIFY(x)
+
+#define __SYCLCOMPAT_WARNING(msg)                                              \
+  __pragma(message(__FILE__                                                    \
+                   "(" __SYCLCOMPAT_TOSTRING(__LINE__) "): warning: " msg))
+
+__SYCLCOMPAT_WARNING("syclcompat is deprecated and the deprecation warnings "
+                     "are ignored unless compiled with /W3 or above.")
+
+#undef __SYCLCOMPAT_WARNING
+#undef __SYCLCOMPAT_TOSTRING
+#undef __SYCLCOMPAT_STRINGIFY
+#endif
```

Comment:
I realized that this isn't of much help as SYCL headers are treated as system headers and warnings are suppressed from them. Another option is to use `#warning` instead of pragma but not all MSVC versions support #warning, as it's a C++23 feature.
@aelovikov-intel suggestions?

Reply:
What warning? On the `namespace syclcompat`? `pragma message` isn't a warning, just a message.

Reply:
Discussed offline. The problem was with `clang-cl`'s imitation of MSVC's option. See https://godbolt.org/z/96aG1K1G8 - clang-cl's interpretation of `/external:W0` is different from that of MSVC and the clang-cl also disables pragma messages with `/external:W0`.
Also, clang-cl, unlike MSVC, do not ignore `[[deprecated]]` attribute and so, I've modified `warnings_deprecated.cpp` test to also run on windows.